RELEASE_PROCESS.md

```diff
@@ -3,86 +3,201 @@
+creating a Minor release.
+
+Whenever a patch release in the form of <major>.<minor>.<patch> is needed, the
+branch `releases/<major>.<minor>` is used to prepare the release.
+The only difference is that now the VERSION file should contain the suffix
+`<major>.<minor>.<patch>-dev`
+(which should have automatically been bumped on the last release).
+
+This means that creating candidates for the patch or creating the release is
+equivalent by using the [`release`](./.github/workflows/release.yaml) GitHub
+Action.
+Make sure that all patched commits have been cherry-picked from main.
+
+*NOTE: It is not valid to create a fresh commit on a patch branch without a
+corresponding cherry-pick from main. Use the next minor version for features.*
+
+#### How to cherry-pick a commit from Main to a Patch branch
+
+To cherry-pick a commit from `main` to a patch branch, the following steps are
+necessary:
+
+1. Checkout the release branch for the patch, e.g. `releases/0.17` for a patch
+   release of `0.17.0` towards `0.17.1`.
+2. Cherry-pick the commit from `main` to the patch branch, e.g.
+   `git cherry-pick <commit-hash>`. Resolve conflicts as necessary.
+3. Create a Pull-Request for the cherry-picked commit to the branch,
+   and prefix it with `[releases/<major>.<minor>]` (in this case
+   `[releases/0.17]`)
+   to signal that this is a patch commit, e.g. with `gh`
+
+   ```shell
+   gh pr create \
+      --title "[releases/0.17] cherry-pick: <Original PR or Commit>" \
+      --body "Cherry-pick of <Original PR or Commit> from main to releases/0.17" \
+      --base releases/0.17 \
+      --draft
+   ```
+
+4. Merge the Pull-Request to the patch branch.
+
+### How Release Notes are managed
+
+For every release branch there is a Github Action
+called [Release Drafter](./.github/workflows/release-drafter.yaml).
+This workflow interprets the Pull Requests merged against main and the release
+branches and generates a draft release
+in Github which can be formed and edited.
+
+Note that when you are updating the branches, the release notes currently get
+overwritten from scratch so any edits get lost.
+In case you want to permanently change the release notes, you will have to carry
+them through all release candidates manually.
+(TODO: this needs improvement by allowing us to do "append-only" style release
+notes)
+
+When a release is created, the release notes are copied to
+the [./docs/releasenotes](./docs/releasenotes) folder in that branch
+with a dedicated commit. If the release notes for that branch already exist in
+the folder, the release notes are taken from the file
+instead of the draft notes from GitHub. This means that you can also create a
+Pull Request to the branch with the corresponding release notes
+and it will be used for the release.
+(TODO: Currently we have not decided if we want to permanently keep the notes
+and thus need a PR to main to append them, or if we want to drop them in the
+future.)
```

Comment:
@morri-son - didn't we agree already on throwing them away?

Reply:
We originally did, but due to the issues we have with release note maintenance we will have to tackle this in a separate issue that likely requires rewriting how we handle the release note generation. I created https://github.com/open-component-model/ocm/issues/1097 to follow up on this specifically because we might even need to change how we generate the notes